plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -38,6 +39,12 @@ const (
 	PluginName = "NamespaceLifecycle"
 	// how long a namespace stays in the force live lookup cache before expiration.
 	forceLiveLookupTTL = 30 * time.Second
+	// how long to wait for a missing namespace before re-checking the cache (and then doing a live lookup)
+	// this accomplishes two things:
+	// 1. It allows a watch-fed cache time to observe a namespace creation event
+	// 2. It allows time for a namespace creation to distribute to members of a storage cluster,
```

Comment:
nit: there is a case 3 which is non-HA, namespace is marked for deletion, and the admission controller local cache is purged.  the next attempt to create a resource in that namespace will wait as well and reduce our flakiness in e2e tests.